src/utilities/tokenizeFieldFilterQuery.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Tokenizes a filter query into its fields and values.
+ *
+ * @param query The filter query (e.g. `"tag:'kuma.io/zone: cluster-1' tag:'version:2' name:cluster"`).
+ * @param allowedFields A list of field names that are allowed. When not empty, a non-existing field reference in the query will cause an error to be thrown.
+ * @returns a list of fields and their values as entries.
+ */
+export function tokenizeFieldFilterQuery(query: string, allowedFields: string[]): [string, string][] {
+  // Prefixes the query with a space in order to easier detect field names.
+  const prefixedQuery = ' ' + query
+  // Matches field names in the query. Field names may contain general word characters and also, spaces and dashes.
+  const fieldMatches = prefixedQuery.matchAll(/ ([-\s\w]+):\s*/g)
+  const fieldEntries: [string, string][] = []
+
+  for (const match of Array.from(fieldMatches)) {
+    if (match.index === undefined) {
+      continue
+    }
+
+    // Normalizes field names. Allows field names to be written in dash or space-delimited form (e.g. “name-prefix” or “name prefix” instead of “name”).
+    const fieldName = getCamelCasedField(match[1])
```

Comment:
Not a blocker at all, and not sure if I understood this right: Just wondering if we are ever going to need to support being able to type `name-prefix`, `namePrefix` or `name prefix` - that's what a lot of this is doing right, supporting all of those? Wouldn't we just use `name` and nothing else?

Reply:
For multi-word strings, we’d likely have camel-cased fields on the API layer. It’s very easy to convert space-delimited and dash-delimited names to camel-cased variants which is nice to have so you can do what you said: type name prefix or name-prefix and have it work.

Now since we renamed `namePrefix` to `name`, this is technically not relevant right now as we don’t have multi-word parameters, but the functionality comes at hardly any extra cost so I’m leaving this in.